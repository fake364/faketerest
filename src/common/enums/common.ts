export enum GENDER_ENUM {
  FEMALE = -1,
  NOT_SPECIFIED = 0,
  MALE
}
