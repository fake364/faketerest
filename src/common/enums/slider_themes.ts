export enum SLIDER_THEMES {
	DINNER_IDEA = 'dinnerIdea',
	HOME_DECOR = 'homeDecor',
	NEW_OUTFIT = 'newOutfit',
	THUMB_IDEA = 'thumbIdea'
}
