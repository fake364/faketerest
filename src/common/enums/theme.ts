export enum THEME_TYPE {
	BASE = 'base',
	BUSINESS = 'business',
	NEWSROOM = 'newsroom'
}