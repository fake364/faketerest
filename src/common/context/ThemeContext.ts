import React from 'react';
import { THEME_TYPE } from '../enums/theme';

const ThemeContext = React.createContext<{
	theme: THEME_TYPE;
	setTheme: Function;
}>({
	theme: THEME_TYPE.BASE,
	setTheme: () => null
});

export default ThemeContext;
