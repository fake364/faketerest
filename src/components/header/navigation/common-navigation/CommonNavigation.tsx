import React, { useContext } from 'react';
import LogoTitle from './logo-title/LogoTitle';
import LoginButtonsContainer from './login-buttons/LoginButtonsContainer';
import ThemeContext from '../../../../common/context/ThemeContext';
import { THEME_TYPE } from '../../../../common/enums/theme';
import BaseNavigationLinks from './links-row/BaseNavigationLinks';

type Props = {};

const CommonNavigation: React.FC<Props> = () => {
  const { theme } = useContext(ThemeContext);
  return (
    <>
      <LogoTitle />
      <LoginButtonsContainer>
        {/*{theme === THEME_TYPE.BASE && <BaseNavigationLinks />} COMMENTED THIS BECAUSE THERE's no functionality*/}
      </LoginButtonsContainer>
    </>
  );
};

export default CommonNavigation;
