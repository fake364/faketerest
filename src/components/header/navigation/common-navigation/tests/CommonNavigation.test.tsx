import React from 'react';
import { mount } from 'enzyme';
import ThemeContext from '../../../../../common/context/ThemeContext';
import { THEME_TYPE } from '../../../../../common/enums/theme';
import CommonNavigation from '../CommonNavigation';

describe('CommonNavigation', () => {
  it('should display logo, and links for BASE theme', () => {
    const component = mount(
      <ThemeContext.Provider value={{ theme: THEME_TYPE.BASE }}>
        <CommonNavigation />
      </ThemeContext.Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
