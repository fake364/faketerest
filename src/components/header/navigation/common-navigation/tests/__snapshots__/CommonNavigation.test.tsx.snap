// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonNavigation should display logo, and links for BASE theme 1`] = `
<CommonNavigation>
  <LogoTitle>
    <div
      className="flex items-center"
    >
      <Image
        alt="Logo"
        height={32}
        src={
          Object {
            "process": [Function],
          }
        }
        width={32}
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              Object {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2732%27%20height=%2732%27/%3e"
              style={
                Object {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <ImageElement
            alt="Logo"
            blurStyle={Object {}}
            config={
              Object {
                "allSizes": Array [
                  16,
                  32,
                  48,
                  64,
                  96,
                  128,
                  256,
                  384,
                  640,
                  750,
                  828,
                  1080,
                  1200,
                  1920,
                  2048,
                  3840,
                ],
                "contentSecurityPolicy": "script-src 'none'; frame-src 'none'; sandbox;",
                "dangerouslyAllowSVG": false,
                "deviceSizes": Array [
                  640,
                  750,
                  828,
                  1080,
                  1200,
                  1920,
                  2048,
                  3840,
                ],
                "disableStaticImages": false,
                "domains": Array [],
                "formats": Array [
                  "image/webp",
                ],
                "imageSizes": Array [
                  16,
                  32,
                  48,
                  64,
                  96,
                  128,
                  256,
                  384,
                ],
                "loader": "default",
                "minimumCacheTTL": 60,
                "path": "/_next/image",
              }
            }
            heightInt={32}
            imgAttributes={
              Object {
                "sizes": undefined,
                "src": "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7",
                "srcSet": undefined,
              }
            }
            imgStyle={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
            isLazy={true}
            isVisible={false}
            layout="intrinsic"
            loader={[Function]}
            onLoadingCompleteRef={
              Object {
                "current": undefined,
              }
            }
            placeholder="empty"
            setBlurComplete={[Function]}
            setIntersection={[Function]}
            srcString=""
            unoptimized={true}
            widthInt={32}
          >
            <img
              alt="Logo"
              data-nimg="intrinsic"
              decoding="async"
              onError={[Function]}
              onLoad={[Function]}
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
            <noscript>
              <img
                alt="Logo"
                data-nimg="intrinsic"
                decoding="async"
                loading="lazy"
                src=""
                style={
                  Object {
                    "border": "none",
                    "bottom": 0,
                    "boxSizing": "border-box",
                    "display": "block",
                    "height": 0,
                    "left": 0,
                    "margin": "auto",
                    "maxHeight": "100%",
                    "maxWidth": "100%",
                    "minHeight": "100%",
                    "minWidth": "100%",
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "padding": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "width": 0,
                  }
                }
              />
            </noscript>
          </ImageElement>
        </span>
      </Image>
      <span
        className="font-bold text-primary text-[20px] ml-[6px]"
        data-theme="base"
      >
        Faketerest
      </span>
    </div>
  </LogoTitle>
  <LoginButtonsContainer>
    <div>
      <BaseNavigationLinks>
        <BaseLink
          className="mr-10 text-blue"
          href="/about"
        >
          <ForwardRef(LinkComponent)
            href="/about"
          >
            <span
              className="mr-10 text-blue text-black cursor-pointer hover:underline font-medium"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              links.base.description
            </span>
          </ForwardRef(LinkComponent)>
        </BaseLink>
        <BaseLink
          className="mr-10"
          href="/business"
        >
          <ForwardRef(LinkComponent)
            href="/business"
          >
            <span
              className="mr-10 text-black cursor-pointer hover:underline font-medium"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              links.base.business
            </span>
          </ForwardRef(LinkComponent)>
        </BaseLink>
        <BaseLink
          className="mr-10"
          href="/blog"
        >
          <ForwardRef(LinkComponent)
            href="/blog"
          >
            <span
              className="mr-10 text-black cursor-pointer hover:underline font-medium"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              links.base.blog
            </span>
          </ForwardRef(LinkComponent)>
        </BaseLink>
      </BaseNavigationLinks>
      <LoginButtons>
        <PrimaryButton
          className="mr-[8px]"
        >
          <BaseButton
            className="mr-[8px] bg-primary px-[12px] py-[8px] hover:bg-primary_hovered text-text_primary rounded-common_radius"
          >
            <button
              className="font-bold active:scale-90 active:brightness-15 mr-[8px] bg-primary px-[12px] py-[8px] hover:bg-primary_hovered text-text_primary rounded-common_radius"
              data-theme="base"
            >
              loginButtons.login
            </button>
          </BaseButton>
        </PrimaryButton>
        <SecondaryButton>
          <BaseButton
            className=" px-[12px] py-[8px] bg-secondary hover:bg-secondary_hovered text-text_secondary rounded-common_radius"
          >
            <button
              className="font-bold active:scale-90 active:brightness-15  px-[12px] py-[8px] bg-secondary hover:bg-secondary_hovered text-text_secondary rounded-common_radius"
              data-theme="base"
            >
              loginButtons.register
            </button>
          </BaseButton>
        </SecondaryButton>
      </LoginButtons>
    </div>
  </LoginButtonsContainer>
</CommonNavigation>
`;
